sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,389 @@ def to_rotation_matrix(self, v=None):
+        same plane, and the real part must be zero for it to make sense.
+
+        Parameters
+        ==========
+
+        q1 : a Quaternion
+        q2 : a Quaternion
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q1 = Quaternion(1, 4, 4, 4)
+        >>> q2 = Quaternion(2, 8, 8, 8)
+        >>> Quaternion.coplanar(q1, q2)
+        True
+
+        >>> q1 = Quaternion(2, 8, 13, 12)
+        >>> Quaternion.coplanar(q1, q2)
+        False
+
+        See Also
+        ========
+
+        is_pure
+
+        """
+        if (q1.a == q1.b == q1.c == q1.d == 0) or (q2.a == q2.b == q2.c == q2.d == 0):
+            raise ValueError('Any of the provided quaternions must not be 0')
+
+        aq = q1.axis()
+        ar = q2.axis()
+
+        return (aq == ar or aq == -ar)
+
+    @classmethod
+    def ternary_coplanar(cls, q1, q2, q3):
+        """
+        Returns True if the axis of the pure quaternions seen as 3D vectors
+        q1, q1, and q2 are coplanar else returns False.
+
+        Explanation
+        ===========
+
+        Two general quaternions are coplanar (in this arc sense) when their axes are parallel.
+
+        Parameters
+        ==========
+
+        q1 : a pure Quaternion.
+        q2 : a pure Quaternion.
+        q3 : a pure Quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q1 = Quaternion(0, 4, 4, 4)
+        >>> q2 = Quaternion(0, 8, 8, 8)
+        >>> q3 = Quaternion(0, 24, 24, 24)
+        >>> Quaternion.ternary_coplanar(q1, q2, q3)
+        True
+
+        >>> q1 = Quaternion(0, 8, 16, 8)
+        >>> q2 = Quaternion(0, 8, 3, 12)
+        >>> Quaternion.ternary_coplanar(q1, q2, q3)
+        False
+
+        """
+
+        if not q2.a == 0 or not q3.a == 0 or not q1.a == 0:
+            raise ValueError('The provided quaternions must be pure')
+
+        M = Matrix([[q1.b, q1.c, q1.d], [q2.b, q2.c, q2.d], [q3.b, q3.c, q3.d]])
+        return M.det() == 0
+
+    def parallel(self, other):
+        """
+        Returns True if the two pure quaternions seen as 3D vectors are parallel else returns False.
+
+        Explanation
+        ===========
+
+        Two pure quaternions are called parallel when their product is commutative.
+
+        Parameters
+        ==========
+
+        other : a Quaternion
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(0, 4, 4, 4)
+        >>> q1 = Quaternion(0, 8, 8, 8)
+        >>> q.parallel(q1)
+        True
+
+        >>> q1 = Quaternion(0, 8, 13, 12)
+        >>> q.parallel(q1)
+        False
+
+        """
+
+        if not self.a == 0 or not other.a == 0:
+            raise ValueError('The provided quaternions must be pure')
+
+        return (Quaternion._generic_mul(self, other) == Quaternion._generic_mul(other, self))
```

Comment:
Why is this using `_generic_mul`? Is that different from `self*other`?

What happens in the symbolic case here?

Probably this should also use three-valued logic like:
```
return (self*other - other*self).is_zero_quaternion
```